site/app/templates/admin/Extensions.twig

```diff
@@ -7,18 +7,27 @@
         <div class="option">
             <p> Use this form to grant an extension (e.g., for an excused absence) to a user on a specific assignment.<br><br><br></p>
             <div class="option"><label for="g_id">Select Gradeable:</label><br>
-                <select name="g_id" id="g_id" onchange="loadHomeworkExtensions($(this).val());" style="margin-top: 10px; width: 50%">
+                <select name="g_id" id="g_id" onchange="loadHomeworkExtensions($(this).val(),$(this).find('option:selected').attr('dueDate'));" style="margin-top: 10px; width: 50%">
                     <option disabled selected value> -- select an option -- </option>
-                    {% for index, value in gradeable_ids %}
-                        <option value="{{ value['g_id'] }}">{{ value['g_title'] }}</option>
+                    {% for index, value in gradeables %}
+                        <option value="{{value['g_id']}}" dueDate="{{value['eg_submission_due_date']}}">{{ value['g_title'] }}</option>
```

Comment:
Use a [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) to contain the due date, not add a custom attribute.